src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/DateTimeRangeRewriter.cs

```diff
@@ -0,0 +1,86 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Features.Search.Expressions;
+using Microsoft.Health.Fhir.ValueSets;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search.Expressions.Visitors
+{
+    /// <summary>
+    /// Rewrites (And (FieldGreaterThanOrEqual DateTimeStart x) (FieldLessThanOrEqual DateTimeStart y)) to
+    /// (And (FieldGreaterThanOrEqual DateTimeStart x) (FieldLessThanOrEqual DateTimeEnd y) (FieldLessThanOrEqual DateTimeStart y)).
+    /// This rewriting constrains the range scan over the index (DateTimeStart, DateTimeEnd).
+    /// </summary>
+    internal class DateTimeRangeRewriter : ExpressionRewriterWithDefaultInitialContext<object>
+    {
+        internal static readonly DateTimeRangeRewriter Instance = new DateTimeRangeRewriter();
+
+        public override Expression VisitSearchParameter(SearchParameterExpression expression, object context)
+        {
+            if ((expression.Parameter.Type == SearchParamType.Date
```

Comment:
I think some comment would help here why this rule exists. Is it because the start/end of LastUpdated will always be the same and therefore don't need the rewrite? Also, is composite here because one of the component of composite could be date?

Reply:
Correct. Adding commments.